spring-rabbit-test/src/test/java/org/springframework/amqp/rabbit/test/ExampleRabbitListenerSpyTest.java

```diff
@@ -0,0 +1,160 @@
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.test.mockito.LatchCountDownAndCallRealMethodAnswer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * @author Gary Russell
+ * @since 1.6
+ *
+ */
+@ContextConfiguration
+@RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
+public class ExampleRabbitListenerSpyTest {
+
+	@Rule
+	public BrokerRunning brokerRunning = BrokerRunning.isRunning();
+
+	@Autowired
+	private RabbitTemplate rabbitTemplate;
+
+	@Autowired
+	private Queue queue1;
+
+	@Autowired
+	private Queue queue2;
+
+	@Autowired
+	private RabbitListenerTestHarness harness;
+
+	@Test
+	public void testTwoWay() throws Exception {
+		assertEquals("FOO", this.rabbitTemplate.convertSendAndReceive(this.queue1.getName(), "foo"));
+
+		Listener listener = this.harness.getSpy("foo");
+		assertNotNull(listener);
+		verify(listener).foo("foo");
+	}
+
+	@Test
+	public void testOneWay() throws Exception {
+		Listener listener = this.harness.getSpy("bar");
+		assertNotNull(listener);
+
+		LatchCountDownAndCallRealMethodAnswer answer = new LatchCountDownAndCallRealMethodAnswer(2);
+		doAnswer(answer).when(listener).foo(anyString(), anyString());
+
+		this.rabbitTemplate.convertAndSend(this.queue2.getName(), "bar");
+		this.rabbitTemplate.convertAndSend(this.queue2.getName(), "baz");
+
+		assertTrue(answer.getLatch().await(10, TimeUnit.SECONDS));
+		verify(listener).foo("bar", this.queue2.getName());
+		verify(listener).foo("baz", this.queue2.getName());
+	}
+
+	@Configuration
+	@RabbitListenerTest
+	public static class Config {
+
+		@Bean
+		public Listener listener() {
+			return new Listener("x");
+		}
+
+		@Bean
+		public ConnectionFactory connectionFactory() {
+			return new CachingConnectionFactory("localhost");
+		}
+
+		@Bean
+		public Queue queue1() {
+			return new AnonymousQueue();
+		}
+
+		@Bean
+		public Queue queue2() {
+			return new AnonymousQueue();
+		}
+
+		@Bean
+		public RabbitAdmin admin(ConnectionFactory cf) {
+			return new RabbitAdmin(cf);
+		}
+
+		@Bean
+		public RabbitTemplate template(ConnectionFactory cf) {
+			return new RabbitTemplate(cf);
+		}
+
+		@Bean
+		public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory(ConnectionFactory cf) {
+			SimpleRabbitListenerContainerFactory containerFactory = new SimpleRabbitListenerContainerFactory();
+			containerFactory.setConnectionFactory(cf);
+			return containerFactory;
```

Comment:
I guess we can't register such a bean by default because of the required `ConnectionFactory`, which should be injected somehow.
But let's introduce the ctor at least to simplify the `SimpleRabbitListenerContainerFactory` `@Bean` definition!

Yes, isn't related to the current PR...